source/core/DualTypeVisitor.cpp

```diff
@@ -61,6 +61,9 @@ namespace Vireo
                         success = ArrayCompatible(typeRefX, typeRefY, operation);
                     }
                     break;
+                case kEncoding_None:
```

Comment:
The `underlyingType` on empty variant constants is of `kEncoding_None`. Treating them as equals in the `TypesCompatible` check and `Apply` cases.

Reply:
I think we need to verify the encoding of typeRefY is also kEncoding_None and return true in that case.